src/Sylius/Bundle/PromotionBundle/Validator/PromotionDateRangeValidator.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\PromotionBundle\Validator;
+
+use Sylius\Component\Promotion\Model\PromotionInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Łukasz Chruściel <lukasz.chrusciel@lakion.com>
+ */
+class PromotionDateRangeValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @param mixed $value
+     * @param Constraint $constraint
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (null === $value) {
+            return;
+        }
+
+        Assert::isInstanceOf(
+            $value,
+            PromotionInterface::class,
+            sprintf('Expecting %s class. %s given', PromotionInterface::class, get_class($value))
```

Comment:
The default message is `'Expected an instance of %2$s. Got: %s'`, so this custom one is not needed.